backend/src/main/java/ai/verta/modeldb/experiment/subtypes/CreateExperimentHandler.java

```diff
@@ -0,0 +1,295 @@
+                    .collect(Collectors.toList()))
+            .setVersionNumber(1L);
+
+    if (request.getDateCreated() != 0L) {
+      experimentBuilder
+          .setDateCreated(request.getDateCreated())
+          .setDateUpdated(request.getDateCreated());
+    } else {
+      experimentBuilder
+          .setDateCreated(Calendar.getInstance().getTimeInMillis())
+          .setDateUpdated(Calendar.getInstance().getTimeInMillis());
+    }
+    if (userInfo != null) {
+      experimentBuilder.setOwner(userInfo.getVertaInfo().getUserId());
+    }
+
+    return experimentBuilder.build();
+  }
+
+  public InternalFuture<Experiment> insertExperiment(Experiment newExperiment) {
+    final var now = Calendar.getInstance().getTimeInMillis();
+    Map<String, Object> valueMap = new LinkedHashMap<>();
+    valueMap.put("id", newExperiment.getId());
+    valueMap.put("project_id", newExperiment.getProjectId());
+    valueMap.put("name", newExperiment.getName());
+    valueMap.put("description", newExperiment.getDescription());
+    valueMap.put("date_created", newExperiment.getDateCreated());
+    valueMap.put("date_updated", newExperiment.getDateUpdated());
+    valueMap.put("owner", newExperiment.getOwner());
+    valueMap.put("version_number", newExperiment.getVersionNumber());
+
+    valueMap.put("deleted", false);
+    valueMap.put("created", false);
+
+    return jdbi.useHandle(
+            handle ->
+                handle.useTransaction(
+                    TransactionIsolationLevel.SERIALIZABLE,
+                    handleForTransaction -> {
+                      Boolean exists =
+                          checkInsertedEntityAlreadyExists(handleForTransaction, newExperiment);
+                      if (exists) {
+                        throw new AlreadyExistsException(
+                            "Experiment '"
+                                + newExperiment.getName()
+                                + "' already exists in database");
+                      }
+
+                      String queryString = buildInsertQuery(valueMap, "experiment");
+
+                      LOGGER.trace("insert experiment query string: " + queryString);
+                      var query = handleForTransaction.createUpdate(queryString);
+
+                      // Inserting fields arguments based on the keys and value of map
+                      for (Map.Entry<String, Object> objectEntry : valueMap.entrySet()) {
+                        query.bind(objectEntry.getKey(), objectEntry.getValue());
+                      }
+
+                      try {
+                        int count = query.execute();
+                        LOGGER.trace("Experiment Inserted : " + (count > 0));
+                      } catch (UnableToExecuteStatementException exception) {
+                        // take a brief pause before resubmitting its query/transaction
+                        Thread.sleep(config.getJdbi_retry_time()); // Time in ms
```

Comment:
This is really, really bad. It adds latency in the whole system, which is exactly what we're trying to optimize for here. When would this exception happen? And can we do a better job than block the whole system?

Reply:
https://github.com/VertaAI/modeldb/pull/2829#discussion_r807760593

Reply:
Removed pause and implemented retry logic using InternalFuture